// Copyright 2015 gRPC authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

option java_multiple_files = true;
option java_package = "io.grpc.relay_commit";
option java_outer_classname = "relay_commit";
option objc_class_prefix = "HLW";

package force_relay_commit;

// The greeting service definition.
service relay_commit {
  // Sends a greeting
  rpc rpc_sendaction (RelayCommitRequest) returns (RelayCommitReply) {}
}

message RelayCommitRequest {
  RelayBlock block = 1;
  repeated RelayAction action = 2;
}

message RelayBlock {
  string producer = 1;
  bytes id = 2;
  bytes previous = 3;
  int32 confirmed = 4;
  bytes transaction_mroot = 5;
  bytes action_mroot = 6;
  bytes mroot = 7;
}

message RelayAction {
  string account = 1;
  string action_name = 2;
  repeated RelayPermission_level authorization = 3;
  bytes data = 4;
}

message RelayPermission_level {
  string actor = 1;
  string permission = 2;
}

// The response message containing the greetings
message RelayCommitReply {
  string reply = 1;
  string message = 2;
}
